cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/conversion/IdentityConversion.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.etl.batch.conversion;
+
+import org.apache.hadoop.io.Writable;
+
+/**
+ * Identity conversion that doesn't convert anything but just casts.
+ *
+ * @param <T> type of object to convert to a Writable
+ * @param <W> the Writable type to convert to
+ */
+public class IdentityConversion<T, W extends Writable> extends WritableConversion<T, W> {
```

Comment:
if it's identity, then why does it need two type parameters? I would think it is:
```
public class IdentityConversion<T extends Writable> extends WritableConversion<T, T>
```